lib/Sema/CSDiag.cpp

```diff
@@ -3195,186 +3058,6 @@ bool FailureDiagnosis::diagnoseNilLiteralComparison(
   return true;
 }
 
-bool FailureDiagnosis::diagnoseArgumentGenericRequirements(
-    TypeChecker &TC, Expr *callExpr, Expr *fnExpr, Expr *argExpr,
-    CalleeCandidateInfo &candidates, ArrayRef<Identifier> argLabels) {
-  if (candidates.closeness != CC_ExactMatch || candidates.size() != 1)
-    return false;
-
-  AbstractFunctionDecl *AFD = nullptr;
```

Comment:
It looks like the old code here didn't handle generic subscripts at all, because it bails out if the candidate is not an AbstractFunctionDecl. Can you add some tests similar to the existing ones for functions, but for generic subscripts, and see if we produce better diagnostics now?

Reply:
Absolutely!

Reply:
I have added a couple of examples in bcd4180f302967155f78dc27708d1db8352561a3